src/ucp/core/ucp_listener.h

```diff
@@ -15,10 +15,14 @@
  * UCP listener
  */
 typedef struct ucp_listener {
-    ucp_worker_iface_t              wiface;  /* UCT iface to listen on */
-    ucp_listener_accept_callback_t  cb;      /* Listen accept callback */
-    void                            *arg;    /* User's arg for the accept callback */
-    uct_worker_cb_id_t              prog_id; /* Slow-path callback */
+    ucp_worker_iface_t                  wiface;  /* UCT iface to listen on */
+    ucp_listener_accept_callback_t      cb;      /* Listen accept callback which
+                                                    creates an endpoint */
+    ucp_listener_accept_addr_callback_t addr_cb; /* Listen accept callback which
```

Comment:
can it be union with "cb"?

Reply:
it will require other flags field and extra logic to determine which callback is set. I would prefer to have 2 pointers with simple check against NULL value until we need something else.

Reply:
so rename the first callback field to accept_cb, to have consistent naming